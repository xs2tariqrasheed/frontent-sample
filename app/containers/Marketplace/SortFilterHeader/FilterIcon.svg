<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
    <g fill="none" fill-rule="evenodd">
        <rect width="18" height="2" x="3" y="4" fill="#2BB573" rx="1"/>
        <rect width="18" height="2" x="3" y="11" fill="#2BB573" rx="1"/>
        <rect width="18" height="2" x="3" y="18" fill="#2BB573" rx="1"/>
        <circle cx="6" cy="5" r="2" fill="#FFF" stroke="#2BB573" stroke-width="2"/>
        <circle cx="12" cy="12" r="2" fill="#FFF" stroke="#2BB573" stroke-width="2"/>
        <circle cx="18" cy="19" r="2" fill="#FFF" stroke="#2BB573" stroke-width="2"/>
    </g>
</svg>
