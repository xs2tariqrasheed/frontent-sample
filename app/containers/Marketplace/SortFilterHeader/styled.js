import styled from 'styled-components';

export const Consent = styled.label`
  display: block;
  margin-bottom: 1rem;
  position: relative;
  padding-left: 1.8rem;
  cursor: pointer;
  text-align: left;
  font-size: 14px;
  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;

    &:checked ~ span {
      background-color: white;
      border: 4px solid #655bf5;
      &:after {
        display: block;
      }
    }
  }

  span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1.4rem;
    width: 1.4rem;
    background: white;
    border: 0.1rem solid #3b38c6;
    border-radius: 4;

    &:after {
      content: '';
      position: absolute;
      display: none;
      left: 4px;
      top: 1px;
      width: 0.4rem;
      height: 0.7rem;
      border: solid #655bf5;
      border-width: 0 0.2rem 0.2rem 0;
      transform: rotate(45deg);
    }
  }

  &:hover {
    input ~ span {
    }
  }
`;
