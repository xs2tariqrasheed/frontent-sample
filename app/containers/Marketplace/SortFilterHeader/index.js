/*
 * SortFilter Header View
 *
 * This is the SortFilter for the marketplace.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Consent } from './styled';
import { updateSort, setP2PGolfersFlag, setItasGolfersFlag } from '../actions';
import FilteringRow from '../../../components/FilteringRow';
import close from '../../../components/PracticeModal/canel-close-purp.svg';

const SortFilterHeader = props => {
  const { sortSetting, modalOpen, setOpenModal } = props;

  return (
    <>
      <button
        type="submit"
        className={`
        ${modalOpen ? 'block' : 'hidden'}
        modal-overlay 
        md:hidden z-20 fixed w-full h-full bg-black opacity-25 top-0 left-0 cursor-pointer`}
        onClick={() => setOpenModal(false)}
      >
        &nbsp;
      </button>
      <div
        className={`
      ${modalOpen ? 'block' : 'hidden'}
      modal
      z-50 
      bg-white 
      fixed 
      m-5 
      p-10 
      md:p-3
      top-0 
      h-full
      left-0
      md:static
      md:m-0
      md:w-auto
      md:h-auto
      md:block
      `}
      >
        <div className="bg-white">
          <div className="container mx-auto">
            <div className="md:grid md:flex md:items-center grid-cols-12 gap-10">
              <div className="col-span-2">
                <header className="md:hidden flex justify-between items-center">
                  <h1 className="mb-0 uppercase text-sm font-light tracking-wide">
                    Filter
                  </h1>
                  <button type="button" onClick={() => setOpenModal(false)}>
                    <img src={close} alt="Close button" />
                  </button>
                </header>
                <hr className="my-3 md:hidden" />
                <Consent>
                  <input
                    id="consent"
                    type="checkbox"
                    checked={props.itasGolferFlag}
                    onChange={() => {
                      const value = props.itasGolferFlag
                        ? 'filter inactive'
                        : 'filter active';

                      window.dataLayer.push({
                        filterStatus: value, // or 'filter active' if set back
                        filterName: 'credit', // enter filter activated/deactivated
                        event: 'set filter',
                      });

                      props.dispatch(setItasGolfersFlag());
                    }}
                  />
                  <span />
                  Credit card available
                </Consent>
                <Consent>
                  <input
                    id="consent"
                    type="checkbox"
                    checked={props.p2pGolferFlag}
                    onChange={() => {
                      const value = props.p2pGolferFlag
                        ? 'filter inactive'
                        : 'filter active';

                      window.dataLayer.push({
                        filterStatus: value, // or 'filter active' if set back
                        filterName: 'ptp', // enter filter activated/deactivated
                        event: 'set filter',
                      });

                      props.dispatch(setP2PGolfersFlag());
                    }}
                  />
                  <span />
                  Peer-to-peer sale
                </Consent>
              </div>
              <hr className="my-3 md:hidden" />
              <div className="lg:grid col-span-9 grid-cols-2 grid-rows-1 gap-5">
                <FilteringRow
                  title="Category"
                  ariaLabel="Filter Golfers by Category"
                  onSelect={key => props.dispatch(updateSort({ key }))}
                  sortSetting={sortSetting}
                  options={[
                    { value: 'tokenid', displayName: 'Token ID' },
                    { value: 'price', displayName: 'Price' },
                    { value: 'powerpeak', displayName: 'Power' },
                    { value: 'accuracypeak', displayName: 'Accuracy' },
                    { value: 'composurepeak', displayName: 'Composure' },
                    { value: 'staminapeak', displayName: 'Stamina' },
                  ]}
                />
                <hr className="my-3 md:hidden" />
                <FilteringRow
                  title="Sort"
                  ariaLabel="Sort Golfers by price"
                  sortSetting={sortSetting}
                  onSelect={order => props.dispatch(updateSort({ order }))}
                  options={[
                    { value: 'desc', displayName: 'High to Low' },
                    { value: 'asc', displayName: 'Low to High' },
                  ]}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

SortFilterHeader.propTypes = {
  dispatch: PropTypes.func,
  p2pGolferFlag: PropTypes.bool,
  itasGolferFlag: PropTypes.bool,
  modalOpen: PropTypes.bool.isRequired,
  setOpenModal: PropTypes.func.isRequired,
  sortSetting: PropTypes.object,
};

function mapProps(store) {
  let p2pFlag = true;
  let itasFlag = true;

  if (store.marketplaceSettings) {
    p2pFlag = store.marketplaceSettings.p2pGolfersFlag;
    itasFlag = store.marketplaceSettings.itasGolfersFlag;
  }

  return {
    sortSetting: store.marketplaceSettings.sortSetting,
    p2pGolferFlag: p2pFlag,
    itasGolferFlag: itasFlag,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

export default connect(
  mapProps,
  mapDispatchToProps,
)(SortFilterHeader);
