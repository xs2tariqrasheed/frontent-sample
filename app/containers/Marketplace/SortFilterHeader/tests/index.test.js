import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter } from 'react-router-dom';
import SortFilterHeader from '../index';
import configureStore from '../../../../configureStore';

describe('<SortFilterHeader />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    let modalOpen = false;
    const {
      container: { firstChild },
    } = render(
      <Provider store={store.store}>
        <BrowserRouter>
          <SortFilterHeader
            modalOpen={modalOpen}
            setOpenModal={() => {
              modalOpen = true;
              return modalOpen;
            }}
          />
        </BrowserRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
