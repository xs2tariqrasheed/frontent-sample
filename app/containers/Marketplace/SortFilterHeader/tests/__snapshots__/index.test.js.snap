// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortFilterHeader /> should render and match the snapshot 1`] = `
<button
  class="
        hidden
        modal-overlay 
        md:hidden z-20 fixed w-full h-full bg-black opacity-25 top-0 left-0 cursor-pointer"
  type="submit"
>
  Â 
</button>
`;
